### PR TITLE
chore: upgrade gcloud cli and install gke auth plugin

### PR DESCRIPTION
**Pull request by Walther**
**What this PR does / why we need it**:
Related to changes outlined in https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke

**Which issue(s) this PR fixes**:

More task details in Asana.

**Special notes for your reviewer**:

### CODE REVIEW

**Review by stefreak — APPROVED** [review]
👍

File: images/circleci-runner/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #### gcloud base image ####
-FROM google/cloud-sdk:390.0.0 as gcloud
```
Comment (stefreak) [inline images/circleci-runner/Dockerfile:2]: I think this one needs a tag bump in `images/circleci-runner/garden.yml` and then also in the `.circleci/config.yml`

**Review by stefreak — CHANGES_REQUESTED** [review]
`images/circleci-runner` needs a version bump (see comment) and for the other images I'm not 100% sure if there are additional, similar TODOs

**Comment by vvagaytsev** [comment]
I've made some extra changes:
* bumped `circleci-runner` image version in https://github.com/garden-io/garden/pull/3431/commits/ad0302fa443bc1a8779758034b3277ba48f4ca38, built and pushed the new image to the Docker hub
* Updated the CircleCI config to use the new image in https://github.com/garden-io/garden/pull/3431/commits/c0995d22672fdb01a37d708f4ee99e45820f2e75
* Updated GCloud SDK version in the CircleCI config in https://github.com/garden-io/garden/pull/3431/commits/ae0da5bb5a1887db6ebe7e7d82067fda36831fc1

A similar GCloud SDK version update was done in #3013 and in its patch PR #3037.

**Review by stefreak — APPROVED** [review]
LGTM 👍

**Pull request merged**